contrib/guix/libexec/build.sh

```diff
@@ -70,12 +69,21 @@ unset CPLUS_INCLUDE_PATH
 unset OBJC_INCLUDE_PATH
 unset OBJCPLUS_INCLUDE_PATH
 
-export LIBRARY_PATH="${NATIVE_GCC}/lib:${NATIVE_GCC_STATIC}/lib"
 export C_INCLUDE_PATH="${NATIVE_GCC}/include"
 export CPLUS_INCLUDE_PATH="${NATIVE_GCC}/include/c++:${NATIVE_GCC}/include"
 export OBJC_INCLUDE_PATH="${NATIVE_GCC}/include"
 export OBJCPLUS_INCLUDE_PATH="${NATIVE_GCC}/include/c++:${NATIVE_GCC}/include"
 
+case "$HOST" in
+    *darwin*)
+        export LIBRARY_PATH="${NATIVE_GCC}/lib"
```

Comment:
Given these are standard ENV vars supported by both compilers, I'm pretty happy to just leave them where they are.

Reply:
Right. But it seems a bit inconsistent. This PR skips static toolchain directory in `LIBRARY_PATH` for the macOS hosts because it is _never_ used there, right. And Objective C/C++ paths are used _only_ for macOS hosts.

Anyway, I'm fine with this PR as it is.

Reply:
> This PR skips static GCC toolchain directory in LIBRARY_PATH for the macOS hosts because it is never used there, right.

It's skiping it because its no-longer available, and checks in our Guix script would fail if we didn't change this. The same is not currently true for the compiler lang env vars. We'll move forward here, and refactor this further when switching to LLD.